spec/lib/salus/report_spec.rb

```diff
@@ -178,6 +178,123 @@ def build_report(report_uri)
+      end
+
+      it 'should make a call to send the txt report for http URI' do
+        url = 'https://nerv.tk3/salus-report'
+        params = { 'salus_report_param_name' => 'report',
+          'additional_params' => { "foo" => "bar", "abc" => "def" } }
+        directive = { 'uri' => url, 'format' => 'yaml', 'post' => params, 'verbose': false }
+        build_report(directive)
+        directive['format'] = 'txt'
+        report = Salus::Report.new(
+          report_uris: [directive],
+          project_name: 'eva00',
+          custom_info: 'test unit'
+        )
+        config = { 'sources' => { 'valid' => ['word'] } }.deep_symbolize_keys
+        report.instance_variable_set(:@config, config)
+
+        stub_request(:post, "https://nerv.tk3/salus-report").with(body: "{\"foo\"=>\"bar\", \"abc"\
+          "\"=>\"def\", \"report\"=>\"==== Salus Scan v2.10.17 for eva00\\n\\n==== Salus Configur"\
+          "ation Files Used:\\n\\n  word\\n\\n\\nOverall scan status: PASSED\\n\\n\\u250C\\u2500"\
+          "\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u252C\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u252C\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u252C\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2510\\n\\u2502 Scanner \\u2502 "\
+          "Running Time \\u2502 Required \\u2502 Passed \\u2502\\n\\u251C\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u253C\\u2500\\u2500\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u253C\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u253C\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2524\\n\\n\\u2514\\u2500\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2534\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2534\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2500\\u2534\\u2500\\u2500\\u2500\\u2500\\"\
+          "u2500\\u2500\\u2500\\u2500\\u2518\"}",
+           headers: { 'Accept' => '*/*',
+            'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+            'Content-Type' => 'text/plain',
+            'User-Agent' => 'Faraday v1.3.0',
+            'X-Scanner' => 'salus' }).to_return(status: 200, body: "", headers: {})
+
+        expect { report.export_report }.not_to raise_error
+      end
+      it 'should make a call to send the sarif report for http URI' do
+        url = 'https://nerv.tk3/salus-report'
+        params = { 'salus_report_param_name' => 'report',
+          'additional_params' => { "foo" => "bar", "abc" => "def" } }
+        directive = { 'uri' => url, 'format' => 'yaml', 'post' => params, 'verbose': false }
```

Comment:
Any reason not to set the format to sarif on line 274 instead of line 276?

Reply:
I was only trying to avoid modifying other tests. the scan_report used in build directive does not have a version and contains duplicate reports. generating a sarif report from this scan report would return an empty report because it would generate an invalid sarif report.
https://github.com/coinbase/salus/blob/02f7678bca84c8c89d03a36059a03373ec746fad/spec/lib/salus/report_spec.rb#L125-L142

Reply:
i was trying to find a work around to avoid using the scan reports from build_reports, the build_reports has duplicate reports. Sarif returns an empty report when there’s duplicate entries